nautobot/core/tests/test_api.py

```diff
@@ -805,16 +815,60 @@ def test_descending_sort(self):
         for field_type, field_name in self.field_type_map.items():
             with self.subTest(f"Testing {field_type}"):
                 response = self.client.get(f"{self.url}?sort=-{field_name}&limit=10", **self.header)
-                self.assertHttpStatus(response, 200)
-                self.assertEqual(
-                    list(map(lambda p: p["id"], response.data["results"])),
-                    list(
-                        map(
-                            lambda p: str(p),  # pylint: disable=unnecessary-lambda
-                            Provider.objects.order_by(f"-{field_name}").values_list("id", flat=True)[:10],
-                        )
-                    ),
-                )
+                self._validate_sorted_response(response, Provider, f"-{field_name}")
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_sorting_tree_node_models(self):
+        location_type = dcim_models.LocationType.objects.get(name="Campus")
+        locations = dcim_models.Location.objects.filter(location_type=location_type)
+        devices = dcim_models.Device.objects.all()
+
+        # When sorting by the 'parent' field, we may encounter different results from API response and ORM filter due to None values.
+        # For instance, sorting tenants on `parent` = `None` might yield either `<Tenant 1 parent=None>, <Tenant 2 parent=None>`
+        # or `<Tenant 2 parent=None>, <Tenant 1 parent=None>`.
+        # To ensure consistent sorting, we'll delete all None parents except one.
```

Comment:
My best guess is this might have something to do with `TreeNode` approach to sorting.

Reply:
That's concerning - ordering should be consistent when using sorting. Let's dig into this further if we can.

Reply:
This actually makes sense - when our `order_by` criterion is something with the same value across multiple objects, we have no guarantee what relative order those same-value objects will be given.